cmd/lhsm-plugin-az-core/archive.go

```diff
@@ -107,9 +108,20 @@ func Archive(ctx context.Context, copier copier.Copier, o ArchiveOptions) (int64
 
 			blob := o.ContainerURL.NewBlockBlobClient(blobpath)
 			_, err = blob.UploadBuffer(ctx, nil, options)
+			if util.ShouldRefreshCreds(err) {
+				o.ContainerURL, blob, err = o.GetNewStorageClients(blobpath)
+				if err != nil {
+					util.Log(pipeline.LogError, fmt.Sprintf("Failed to get new block blob client for %s: %v", blobpath, err))
+					return
+				}
+				blob.UploadBuffer(ctx, nil, options)
```

Comment:
Any risk here that not storing err like we do on 110 can result in the func exiting as if it succeeded when it actually failed?  I'll be honest, Go here has me completely lost on how err gets persisted whatsoever in this goroutine.